localtypings/pxtcustomeditor.d.ts

```diff
@@ -0,0 +1,186 @@
+declare namespace pxt.extension {
+    interface Message {
+        id?: string;
+        type: "request" | "response" | "event";
+    }
+
+    interface Request extends Message {
+        type: "request";
+        subtype: string;
+        
+        seq: number;
+        body?: any;
+    }
+
+    interface Response extends Message {
+        type: "response";
+        subtype: string;
+
+        requestSeq: number;
+        success: boolean;
+        message?: string;
+        body?: any;
+    }
+
+    interface Event extends Message {
+        type: "event";
+        event: string;
+        body?:  any;
+    }
+
+    interface DataStreams<T> {
```

Comment:
Just a generic so that I can reuse this interface for both the request and response (see below)